src/spatial/utils.jl

```diff
@@ -0,0 +1,219 @@
+"""
+A file with types, structs and functions for spatial simulations
+"""
+
+"""
+stores info for a spatial jump
+"""
+struct SpatialJump{J}
+    site::J #where the jump happens
+    index::Int #diffusion or reaction
+    target_site::J #target of diffusive hop
+end
+
+############ abstract spatial system struct ##################
+"""
+Contains all info about the topology of the system
+"""
+abstract type AbstractSpatialSystem end
+
+"""
+returns neighbors of site
+"""
+function neighbors end
+
+"""
+returns total number of sites
+"""
+function number_of_sites end
+
+"""
+return the number of neighbors of a site
+"""
+function num_neighbors end
+
+# TODO make a graph struct with connectivity list, compare with CartesianGrid
+
+################### CartesianGrid <: AbstractSpatialSystem ########################
+#TODO store the number of neighbors for each site or store all neighbors for each site.
+struct CartesianGrid <: AbstractSpatialSystem
+    dimension::Int # dimension of the grid, e.g. 1, 2 or 3
+    linear_size::Int #side length of the grid
+end
```

Comment:
Before we go too much further I'd suggest we add support for rectangular grids and make this a vector. Otherwise call this SquareCartesianGrid.